src/DonorPerfect.php

```diff
@@ -250,8 +255,21 @@ public function call(string $action, array $parameters)
      */
     public function callSql($sql)
     {
+        // Clean the sql of extra spaces and tabs
+        $cleansql = $sql;
+        $done = false;
+        do {
+          $cleansql_new = trim(str_ireplace(["\n", "\t", '  ', '  )'], [' ', '', ' ', ' )'], $cleansql));
+          if ($cleansql_new === $cleansql) {
+            $done = true;
+          }
+          else {
+            $cleansql = $cleansql_new;
+          }
+        } while (!$done);
+
```

Comment:
```suggestion
        // Remove all formatting whitespace while leaving whitespace that is part of value strings
        $in_quote = false;
        $output = '';

        for ($i = 0; $i < strlen($sql); $i++) {
            if ($sql[$i] == "'" || $sql[$i] == '"') {
                $in_quote = !$in_quote;
            }
            if (($sql[$i] == ' ' || $sql[$i] == "\n" || $sql[$i] == "\r") && !$in_quote) {
                if (empty($output) || substr($output, -1) == ' ') {
                    continue;
                }
                $output .= ' ';
            } else {
                $output .= $sql[$i];
            }
        }
        
```

Reply:
@RonAbarbanel what do you think of the above? Kindly provided by ChatGPT after about 15 minutes of back and forth, it takes

```php
$sql = "SELECT *
       FROM my_table
       WHERE col1 = 'val1
                   val1_cont
                   val1_cont'
         AND col2 = \"val2
                   val2_cont
                   val2_cont\"";
```

and turns it into 

```sql
SELECT * FROM my_table WHERE col1 = 'val1
                   val1_cont
                   val1_cont' AND col2 = "val2
                   val2_cont
                   val2_cont"
```

whereas the same input using the do while loop resulted in

```sql
SELECT * FROM my_table WHERE col1 = 'val1 val1_cont val1_cont' AND col2 = "val2 val2_cont val2_cont"
```

Reply:
@jaxwilko @bennothommo FYI my first use of ChatGPT on a programming problem lol 😂

Reply:
Haha, this is fantastic. I did not consider the in-quote possibility. I guess ChatGPT is smarter than me. I'll have to start using it!